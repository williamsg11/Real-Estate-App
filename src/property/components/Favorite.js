import React, {Component, PropTypes} from 'react';
import {StyleSheet, TouchableHighlight, View} from 'react-native';
import FontAwesome from 'react-native-vector-icons/FontAwesome';

export default class Favorite extends Component {
  static propTypes = {
    handleFavoritePress: PropTypes.func.isRequired,
    isFavorited: PropTypes.bool.isRequired,
  };

  render() {
    const {handleFavoritePress, isFavorited} = this.props;
    return (
      <View style={styles.container}>
        <TouchableHighlight
          onPress={() => {
            handleFavoritePress();
          }}
          underlayColor="transparent">
          <FontAwesome
            name={isFavorited ? 'star' : 'star-o'}
            size={25}
            style={[styles.icon]}
          />
        </TouchableHighlight>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  title: {
    color: 'white',
    fontSize: 15,
    paddingLeft: 5,
  },
  icon: {
    width: 25,
    height: 25,
    color: 'gold',
  },
});
