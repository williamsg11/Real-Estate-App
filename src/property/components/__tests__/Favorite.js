import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Favorite from './../Favorite';

test('renders heart', () => {
  const tree = renderer
    .create(<Favorite isFavorited={true} handleFavoritePress={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
