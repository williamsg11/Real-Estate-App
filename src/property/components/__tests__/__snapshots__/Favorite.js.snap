// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heart 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "flex-end",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 25,
          },
          Array [
            Object {
              "color": "gold",
              "height": 25,
              "width": 25,
            },
          ],
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï€…
    </Text>
  </View>
</View>
`;
